File: codegen/results/v002/qwen3.8_27b_think/20260827_171807/go/001_csv_to_json/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"fmt"
	"os"
	"strconv"
)

type Record struct {
	Name  string  `json:"Name"`
	Age   int     `json:"Age"`
	Email string  `json:"Email"`
	Score float64 `json:"Score"`
}

func main() {
	file, err := os.Open("input/data.csv")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error opening file: %v\n", err)
		os.Exit(1)
	}
	defer file.Close()

	reader := csv.NewReader(file)
	records, err := reader.ReadAll()
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error reading CSV: %v\n", err)
		os.Exit(1)
	}

	result := make([]Record, 0)
	for i, row := range records {
		if i == 0 {
			continue // skip header
		}
		age, err := strconv.Atoi(row[1])
		if err != nil {
			fmt.Fprintf(os.Stderr, "Error parsing age: %v\n", err)
			os.Exit(1)
		}
		score, err := strconv.ParseFloat(row[3], 64)
		if err != nil {
			fmt.Fprintf(os.Stderr, "Error parsing score: %v\n", err)
			os.Exit(1)
		}
		result = append(result, Record{
			Name:  row[0],
			Age:   age,
			Email: row[2],
			Score: score,
		})
	}

	output, err := json.Marshal(result)
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error marshaling JSON: %v\n", err)
		os.Exit(1)
	}

	fmt.Println(string(output))
}

File: codegen/results/v002/qwen3.8_27b_think/20260827_171807/typescript/001_csv_to_json/solution.ts
import * as fs from 'fs';
import * as path from 'path';

const filePath = path.resolve('input/data.csv');
const content = fs.readFileSync(filePath, 'utf-8');
const lines = content.trim().split('\n');

const header = lines[0].split(',');
const result = lines.slice(1).map(line => {
  const cols = line.split(',');
  const obj: Record<string, any> = {};
  for (let i = 0; i < header.length; i++) {
    const key = header[i].trim();
    const value = cols[i].trim();
    if (key === 'Age') {
      obj[key] = parseInt(value, 10);
    } else if (key === 'Score') {
      obj[key] = parseFloat(value);
    } else {
      obj[key] = value;
    }
  }
  return obj;
});

process.stdout.write(JSON.stringify(result, null, 2) + '\n');